src/result.rs

```diff
@@ -147,530 +158,32 @@ impl<T, E: std::error::Error, S: fmt::Display> Termination for Result<T, E, S> {
  *  FIGLET: impl Result
  */
 
-/// Stuff not from the standard library.
-impl<T, E, S: Traced + Default> Result<T, E, S> {
-    /// Constructs a new error result from the provided error value.
-    ///
-    /// # Examples
-    ///
-    /// Basic usage:
-    ///
-    /// ```
-    /// let x: Result<u32, &str> = Result::new_err("Nothing here");
-    /// ```
-    #[inline]
-    #[track_caller]
-    pub fn new_err<D>(error_value: D) -> Self
-    where
-        E: From<D>,
-    {
-        Err(TracedError::new(E::from(error_value)))
-    }
-}
-
-impl<T, E, S: Traced> Result<T, E, S> {
-    /// Converts from `Result<T, E>` to [`std::result::Result<T, E>`].
-    ///
-    /// Converts `self` into a [`std::result::Result<T, E>`], consuming `self`.
-    ///
-    /// # Examples
-    ///
-    /// Basic usage:
-    ///
-    /// ```
-    /// let x: Result<u32, &str> = Ok(2);
-    /// assert_eq!(x.to_std(), std::result::Result::Ok(2));
-    ///
-    /// let x: Result<u32, &str> = Result::new_err("Nothing here");
-    /// assert_eq!(x.to_std(), std::result::Result::Err("Nothing here"));
-    /// ```
-    #[inline]
-    pub fn to_std(self) -> std::result::Result<T, E> {
-        match self {
-            Ok(t) => std::result::Result::Ok(t),
-            Err(e) => std::result::Result::Err(e.error),
-        }
-    }
-
-    #[inline]
-    pub fn as_std_ref(&self) -> std::result::Result<&T, &E> {
-        match self {
-            Ok(ref t) => std::result::Result::Ok(t),
-            Err(ref e) => std::result::Result::Err(&e.error),
-        }
-    }
-
-    /// Converts from `Result<T, E>` to [`Option<TracedError<E>>`].
-    ///
-    /// Converts `self` into an [`Option<TracedError<E>>`], consuming `self`,
-    /// and discarding the success value, if any.
-    ///
-    /// # Examples
-    ///
-    /// Basic usage:
-    ///
-    /// ```
-    /// let x: Result<u32, &str> = Ok(2);
-    /// assert_eq!(x.err_stack(), None);
-    ///
-    /// let x: Result<u32, &str> = Result::new_err("Nothing here");
-    /// match x.err_stack() {
-    ///     Some(e) => assert_eq!(*e, "Nothing here"),
-    ///     None => unreachable!(),
-    /// }
-    /// ```
-    #[inline]
-    pub fn err_stack(self) -> Option<TracedError<E, S>> {
-        match self {
-            Ok(_) => None,
-            Err(x) => Some(x),
-        }
-    }
-}
-
-impl<T, E> Result<T, E> {
-    /////////////////////////////////////////////////////////////////////////
-    // Querying the contained values
-    /////////////////////////////////////////////////////////////////////////
```

Comment:
If needed, we can add back in certain bits of functionality from `std::result::Result`. For example, having the following methods on `propagate::Result` could prove useful:

* `map_err()`
* `err()`
* `ok()`
* `unwrap()`
* `unwrap_or_else()`

Point being, nothing about this change prevents us from making the interface of `propagate::Result` closer to `std::result::Result` down the line if we feel that it is called for.

I'll spend some time testing the ergonomics of `propagate::Result` at work over the coming weeks.